src/python_pachyderm/mixin/pps.py

```diff
@@ -602,3 +686,35 @@ def garbage_collect(self, memory_bytes=None):
         precise garbage collection (at the cost of more memory usage).
         """
         return self._req(Service.PPS, "GarbageCollect", memory_bytes=memory_bytes)
+
+
+def pipeline_input_name(i):
+    if i is None:
+        return None
+    if i.pfs is not None:
+        return i.pfs.name
+    if i.cross is not None:
+        if len(i.cross) > 0:
+            return pipeline_input_name(i.cross[0])
```

Comment:
Just to check that I understand how this code works, technically you could return the empty string here, right? Like, IIUC, the idea is that you're assembling an iterator over all inputs (`pipeline_inputs` below) and then converting to a list of pipeline names, but only the names of the PFS inputs matter because that's all you check for above?

Reply:
This is just transliterated code from go (I think in `ppsutil`), but looking it over I think it's wrong. I need to iterate through all of the `cross` inputs right?